power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/short_circuit_solver.hpp

```diff
@@ -71,108 +71,111 @@ class ShortCircuitSolver {
                 mat_data_[diagonal_position] += y_source;  // add y_source to the diagonal of Ybus
-                rhs[bus_number] +=
-                    y_source * input.source[source_number] * source_voltage_ref;  // Y_source * U_source * c
+                output.u_bus[bus_number] +=
+                    dot(y_source, ComplexValue<sym>{input.source[source_number] *
+                                                    source_voltage_ref});  // rhs += Y_source * U_source * c
             }
             // add all faults
             for (Idx fault_number = fault_bus_indptr[bus_number]; fault_number != fault_bus_indptr[bus_number + 1];
                  ++fault_number) {
-                DoubleComplex y_fault = input.faults[fault_number].y_fault;
+                const DoubleComplex y_fault = input.faults[fault_number].y_fault;
                 if (std::isinf(y_fault.real())) {
-                    assert(std::isinf(y_fault.imag());
+                    assert(std::isinf(y_fault.imag()));
                     zero_fault_counter[bus_number] += 1;
                     if constexpr (sym) {  // three phase fault
                         for (Idx data_index = y_bus.row_indptr_lu()[bus_number];
                              data_index != y_bus.row_indptr_lu()[bus_number + 1]; ++data_index) {
-                            Idx row_number = y_bus.col_indices_lu()[data_index];
-                            Idx col_data_index = y_bus.lu_transpose_entry()[data_index];
+                            const Idx row_number = y_bus.col_indices_lu()[data_index];
+                            const Idx col_data_index = y_bus.lu_transpose_entry()[data_index];
                             // mat_data[:,bus] = 0
                             // mat_data[bus,bus] = -1
                             if (row_number != bus_number) {
-                                mat_data[col_data_index] = 0;
+                                mat_data_[col_data_index] = 0;
                             }
                             else {
                                 mat_data_[col_data_index] = -1;
                             }
                         }
-                        rhs[bus_number] = 0;
+                        output.u_bus[bus_number] = 0;  // update rhs
                     }
                     else if (fault_type == FaultType::single_phase_to_ground) {
                         for (Idx data_index = y_bus.row_indptr_lu()[bus_number];
                              data_index != y_bus.row_indptr_lu()[bus_number + 1]; ++data_index) {
-                            Idx row_number = y_bus.col_indices_lu()[data_index];
-                            Idx col_data_index = y_bus.lu_transpose_entry()[data_index];
+                            const Idx row_number = y_bus.col_indices_lu()[data_index];
+                            const Idx col_data_index = y_bus.lu_transpose_entry()[data_index];
```

Comment:
here and in other locations of this file, to keep consistency with the rest of the codebase, please use `T const` instead of `const T` and `T const&` instead of `const T&`

Reply:
done